src/components/Courses/Courses.jsx

```diff
@@ -1,45 +1,113 @@
-import React, { useState, useMemo } from 'react';
-import './Courses.css';
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import SearchBar from './components/SearchBar/SearchBar';
 import CourseCard from './components/CourseCard/CourseCard';
 import Button from '../../common/Button/Button';
+import CourseInfo from '../CourseInfo/CourseInfo';
+import { api_endpoints, storageKeys, paths } from '../../constants';
+import './Courses.css';
 
-const Courses = ({ courses, authors, onAddCourseClick }) => {
+const Courses = ({ onAddCourseClick }) => {
+  const [courses, setCourses] = useState([]);
+  const [authors, setAuthors] = useState([]);
   const [searchTerm, setSearchTerm] = useState('');
+  const [selectedCourse, setSelectedCourse] = useState(null);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const token = localStorage.getItem(storageKeys.userToken);
+    if (token) {
+      navigate(paths.courses);
+    }
+    fetchCourses();
+    fetchAuthors();
+  }, [navigate]);
 
-  const handleSearch = (term) => {
+  const fetchCourses = async (query = '', type = '') => {
+    try {
+      let response;
+      if (query!=='' && type!=='') {
```

Comment:
Please run prettier / eslint through this file as it seems to not have all the correct code formatting